flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/config/StartupMode.java

```diff
@@ -19,6 +19,8 @@
 
 import org.apache.flink.annotation.Internal;
 import org.apache.flink.streaming.connectors.kafka.internals.KafkaTopicPartitionStateSentinel;
+import org.apache.flink.streaming.connectors.kafka.table.KafkaConnectorOptions.ScanStartupMode;
+import org.apache.flink.table.api.TableException;
 
 /** Startup modes for the Kafka Consumer. */
 @Internal
```

Comment:
This enum is basically a duplicate of `ScanStartupMode` now. Ideally I'd just merge them, but this would expose the `stateSentinel` stuff in the API. Is that OK or should we keep them separate?

Reply:
I support your current approach. StartupMode is a lower level / runtime class. We should not mix it with API classes.